torchtune/models/phi4/_model_builders.py

```diff
@@ -0,0 +1,120 @@
+from typing import List, Optional
+
+from torchtune.models.phi3._component_builders import phi3, lora_phi3
+from torchtune.models.phi4._tokenizer import Phi4MiniTokenizer
+
+from torchtune.modules import TransformerDecoder
+from torchtune.modules.peft import LORA_ATTN_MODULES
+from functools import partial
+from torchtune.modules.tokenizers import parse_hf_tokenizer_json
+from torchtune.data._prompt_templates import _TemplateType
+from torchtune.data._prompt_templates import _get_prompt_template
+
+
+"""
+Model builders build specific instantiations using component builders. For example
+the ``phi4_14b`` model builder uses the ``phi3`` component builder.
+"""
+
+
+def phi4_14b() -> TransformerDecoder:
+    """
+    Builder for creating the Phi4 (14B) Instruct Model.
+
+    Returns:
+        TransformerDecoder: Instantiation of Phi4 (14B) Instruct Model
+    """
+    return phi3(
+        vocab_size=100_352,
+        num_layers=40,
+        num_heads=40,
+        num_kv_heads=10,
+        embed_dim=5120,
+        intermediate_dim=17920,
+        max_seq_len=16384,
+        attn_dropout=0.0,
+        norm_eps=1e-5,
+    )
+
+def phi4_14b_tokenizer(vocab_path: str = None, merges_path: str = None, path: str = None, special_tokens_path: Optional[str] = None, max_seq_len: Optional[int] = None, prompt_template: Optional[_TemplateType] = None) -> Phi4MiniTokenizer:
+    """Phi4 (14B) tokenizer.
+    Args:
+        path (str): Path to the tiktoken tokenizer model.
```

Comment:
I think the docstring needs to be updated? (Also still not sure why this isn't linting on CI)